src/keylessBackup/encryption.ts

```diff
@@ -0,0 +1,105 @@
+import * as secp from '@noble/secp256k1'
+import crypto from 'crypto'
+import hkdf from 'futoin-hkdf'
+
+/**
+ * Derives a 256-bit key using the HKDF method from two key shares.
+ *
+ * @param {string} keyshare1 - The first keyshare in utf8 format.
+ * @param {string} keyshare2 - The second keyshare in utf8 format.
+ * @returns {Buffer} A derived 256-bit key as a Buffer.
+ */
+export function deriveKeyFromKeyShares(
+  keyshare1: Buffer,
+  keyshare2: Buffer,
+  bytes: number = 32
+): Buffer {
+  // sanity check
+  if (keyshare1.length < 16 || keyshare2.length < 16) {
+    throw new Error('Key shares must be at least 16 bytes long')
+  }
+
+  // Combining the keyshares
+  const combinedKeyShares = Buffer.concat([
+    keyshare1,
+    hkdf(keyshare2, bytes, {
+      salt: 'some fixed salt',
+      info: 'valora.keylessBackup.deriveKeyFromKeyShares',
+      hash: 'SHA-256',
+    }),
+  ])
+
+  // Using futoin-hkdf to derive a 256-bit key
+  return hkdf(combinedKeyShares, bytes, {
+    salt: 'some fixed salt',
+    info: 'valora.keylessBackup.deriveKeyFromKeyShares',
+    hash: 'SHA-256',
+  })
+}
+
+/**
+ * Derives a 256-bit key using the HKDF method from two key shares then derives a secp256k1 pair from that.
+ *
+ * @param {string} keyshare1 - The first keyshare in utf8 format.
+ * @param {string} keyshare2 - The second keyshare in utf8 format.
+ * @returns {Buffer} A derived 256-bit key as a Buffer.
+ */
+export function getSecp256K1KeyPair(keyshare1: Buffer, keyshare2: Buffer): Uint8Array {
+  // Creating two keys and concatonating them to form a 48 byte key since 40 is the minimum for hashToPrivateKey
+  const derivedKey1 = deriveKeyFromKeyShares(keyshare1, keyshare2)
+  const derrivedKey2 = deriveKeyFromKeyShares(keyshare1, keyshare2, 16)
```

Comment:
HDKF can generate longer outputs. Calling with this a length of 48 should work. Also, this will not give a psuedorandom 48-bytes string, the last 16 bytes will be equal to the first 16 bytes.
```suggestion
  const derivedKey1 = deriveKeyFromKeyShares(keyshare1, keyshare2, 48)
```